### PR TITLE
nsimd: New version 3.0

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nsimd/package.py [hunk 1]
```diff
@@ -27,6 +28,7 @@ class Nsimd(CMakePackage):
                 'SSE2', 'SSE42', 'AVX', 'AVX2', 'AVX512_KNL', 'AVX512_SKYLAKE',
                 'NEON128', 'AARCH64',
                 'SVE', 'SVE128', 'SVE256', 'SVE512', 'SVE1024', 'SVE2048',
+                'VMX', 'VSX',
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nsimd/package.py:31]: This LGTM. Wondering if you thought about removing this variant and having SIMD activated based on target inspection:
```python
if 'avx' in self.spec.target:
    ...
```
Maybe we already discussed this sometimes ago in a PR?

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/nsimd/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/nsimd/package.py:31]: Right! I thought about that, and then I erred in thinking that Spack would look at the host architecture. Yes, it should definitively use the target architecture by default.

Comment (eschnett) [reply var/spack/repos/builtin/packages/nsimd/package.py:31]: I don't think there is a feature flag for e.g. `VMX` for IBM POWER. I'd like to enable it whenever there is a POWER CPU without `vsx` feature. How do I do that? Is it `self.microarchitecture in ['ppc64', 'ppc64le']`?

Comment (eschnett) [reply var/spack/repos/builtin/packages/nsimd/package.py:31]: See https://github.com/spack/spack/pull/26850 .

**Pull request merged**